src/Microsoft.Dnx.Runtime.Internals/RuntimeEnvironmentExtensions.cs

```diff
@@ -27,5 +30,45 @@ public static string GetShortVersion(this IRuntimeEnvironment env)
         {
             return $"{env.RuntimeType}-{env.RuntimeArchitecture}-{env.RuntimeVersion}";
         }
+
+        public static string GetRuntimeIdentifier(this IRuntimeEnvironment env)
+        {
+            if (string.Equals(env.RuntimeType, RuntimeTypes.Mono, StringComparison.Ordinal))
+            {
+                return "mono";
+            }
+
+            string os = env.OperatingSystem;
+            string ver = env.OperatingSystemVersion;
+            if (string.Equals(os, RuntimeOperatingSystems.Windows, StringComparison.Ordinal))
+            {
+                os = "win";
+
+                // Convert 6.x to the correct branding version for the RID
+                var parsedVersion = Version.Parse(ver);
+                if(parsedVersion.Major == 6 && parsedVersion.Minor == 1)
```

Comment:
I actually thought would come from native code where I would use IsVersionXOrHiger API to get the version instead of passing what GetVersionEx returns...